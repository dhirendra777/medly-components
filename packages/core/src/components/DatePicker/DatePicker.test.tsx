import { cleanup, fireEvent, render } from '@test-utils';
import React from 'react';
import { DatePicker } from './DatePicker';

describe('DatePicker component', () => {
    afterEach(cleanup);

    it('should render properly when value is null', () => {
        const { container } = render(
                <DatePicker disabled label="Start Date" value={null} displayFormat="MM/dd/yyyy" onChange={jest.fn()} />
            ),
            inputEl = container.querySelector('#medly-datepicker-input') as HTMLInputElement;
        expect(inputEl.value).toEqual('');
    });

    it('should render properly when value is of date type', () => {
        const { container } = render(
                <DatePicker disabled label="Start Date" value={new Date(2020, 0, 1)} displayFormat="MM/dd/yyyy" onChange={jest.fn()} />
            ),
            inputEl = container.querySelector('#medly-datepicker-input') as HTMLInputElement;
        expect(inputEl.value).toEqual('01/01/2020');
        expect(container).toMatchSnapshot();
    });

    it('should render properly when value is of string type', () => {
        const { container } = render(
                <DatePicker disabled label="Start Date" value="01/01/2020" displayFormat="MM/dd/yyyy" onChange={jest.fn()} />
            ),
            inputEl = container.querySelector('#medly-datepicker-input') as HTMLInputElement;
        expect(inputEl.value).toEqual('01/01/2020');
        expect(container).toMatchSnapshot();
    });

    it('should render input as read only', () => {
        const { container } = render(
            <DatePicker disabled label="Start Date" value="01/01/2020" displayFormat="MM/dd/yyyy" onChange={jest.fn()} />
        );
        const inputEl = container.querySelector('#medly-datepicker-input') as HTMLInputElement;
        fireEvent.change(inputEl, { target: { value: 'Dummy' } });
        expect(inputEl.value).toEqual('01/01/2020');
    });

    it('should show calendar on click on input', () => {
        const { container } = render(<DatePicker value={new Date(2020, 0, 1)} displayFormat="MM/dd/yyyy" onChange={jest.fn()} />);
        fireEvent.click(container.querySelector('input'));
        expect(container.querySelector('#medly-datepicker-popover')).toBeVisible();
    });

    it('should hide calendar on click outside of the component', async () => {
        const { container, getByText } = render(
            <>
                <div>sibling</div>
                <DatePicker value={new Date(2020, 0, 1)} displayFormat="MM/dd/yyyy" onChange={jest.fn()} />
            </>
        );
        fireEvent.click(container.querySelector('input'));
        expect(container.querySelector('#medly-datepicker-popover')).toBeVisible();
        fireEvent.click(getByText('sibling'));
        expect(container.querySelector('#medly-datepicker-popover')).toBeNull();
    });

    it('should call onChange on selecting date', async () => {
        const mockOnChange = jest.fn(),
            dateToSelect = new Date(2020, 0, 2),
            { container, getByTitle } = render(
                <DatePicker value={new Date(2020, 0, 1)} displayFormat="MM/dd/yyyy" onChange={mockOnChange} />
            );
        fireEvent.click(container.querySelector('input'));
        fireEvent.click(getByTitle(dateToSelect.toDateString()));
        expect(mockOnChange).toHaveBeenCalledWith(dateToSelect);
    });
});
