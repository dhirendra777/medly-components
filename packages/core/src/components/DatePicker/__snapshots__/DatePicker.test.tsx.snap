// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker component should render properly when value is of date type 1`] = `
.c1 {
  font-size: 14px;
  font-weight: 400;
  color: #616161;
  margin: 0px 8px 0px 0px;
  cursor: default;
}

.c2 {
  grid-area: label;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  margin: 8px 8px 8px 0px;
  display: inline-grid;
  grid-template-columns: minmax(max-content,25%) minmax(auto,80%);
  grid-template-areas: 'label field' 'random description';
}

.c0 .c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  margin: 8px 8px 8px 0px;
  display: inline-grid;
  grid-template-areas: 'label' 'field' 'description';
  grid-template-columns: auto;
  grid-template-areas: 'field' 'description';
}

.c4 .c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-clip: padding-box;
  box-sizing: border-box;
  border: 1px solid #E0E0E0;
  border-radius: 5px;
  overflow: hidden;
}

.c5:focus-within {
  border-color: #012040;
}

.c5 > * {
  padding: 8px;
  font-size: 14px;
  font-weight: 400;
  color: #616161;
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  outline: 0;
  border: 0;
  cursor: not-allowed;
  background-color: #E0E0E0;
}

.c6::-webkit-input-placeholder {
  color: #9E9E9E;
}

.c6::-moz-placeholder {
  color: #9E9E9E;
}

.c6:-ms-input-placeholder {
  color: #9E9E9E;
}

.c6::placeholder {
  color: #9E9E9E;
}

.c6::-webkit-file-upload-button {
  visibility: hidden;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
}

.c3 .c9 {
  top: calc(100% + 5px);
  left: 0%;
}

.c7 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: not-allowed;
}

<div>
  <div
    class="c0"
    id="medly-datepicker"
  >
    <label
      class="c1 c2"
      for="medly-datepicker"
    >
      Start Date
    </label>
    <div
      class="c3"
      id="medly-popover-wrapper"
    >
      <div
        class="c4"
        id="medly-datepicker-input-field"
      >
        
        <div
          class="c5"
        >
          <input
            class="c6 c7"
            disabled=""
            id="medly-datepicker-input"
            placeholder="Select Date"
            type="text"
            value="01/01/2020"
          />
        </div>
        
      </div>
    </div>
  </div>
</div>
`;

exports[`DatePicker component should render properly when value is of string type 1`] = `
.c1 {
  font-size: 14px;
  font-weight: 400;
  color: #616161;
  margin: 0px 8px 0px 0px;
  cursor: default;
}

.c2 {
  grid-area: label;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  margin: 8px 8px 8px 0px;
  display: inline-grid;
  grid-template-columns: minmax(max-content,25%) minmax(auto,80%);
  grid-template-areas: 'label field' 'random description';
}

.c0 .c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  margin: 8px 8px 8px 0px;
  display: inline-grid;
  grid-template-areas: 'label' 'field' 'description';
  grid-template-columns: auto;
  grid-template-areas: 'field' 'description';
}

.c4 .c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-clip: padding-box;
  box-sizing: border-box;
  border: 1px solid #E0E0E0;
  border-radius: 5px;
  overflow: hidden;
}

.c5:focus-within {
  border-color: #012040;
}

.c5 > * {
  padding: 8px;
  font-size: 14px;
  font-weight: 400;
  color: #616161;
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  outline: 0;
  border: 0;
  cursor: not-allowed;
  background-color: #E0E0E0;
}

.c6::-webkit-input-placeholder {
  color: #9E9E9E;
}

.c6::-moz-placeholder {
  color: #9E9E9E;
}

.c6:-ms-input-placeholder {
  color: #9E9E9E;
}

.c6::placeholder {
  color: #9E9E9E;
}

.c6::-webkit-file-upload-button {
  visibility: hidden;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
}

.c3 .c9 {
  top: calc(100% + 5px);
  left: 0%;
}

.c7 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: not-allowed;
}

<div>
  <div
    class="c0"
    id="medly-datepicker"
  >
    <label
      class="c1 c2"
      for="medly-datepicker"
    >
      Start Date
    </label>
    <div
      class="c3"
      id="medly-popover-wrapper"
    >
      <div
        class="c4"
        id="medly-datepicker-input-field"
      >
        
        <div
          class="c5"
        >
          <input
            class="c6 c7"
            disabled=""
            id="medly-datepicker-input"
            placeholder="Select Date"
            type="text"
            value="01/01/2020"
          />
        </div>
        
      </div>
    </div>
  </div>
</div>
`;
